addons/binding/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/internal/connector/UplinkWebInterface.java

```diff
@@ -0,0 +1,188 @@
+ */
+package org.openhab.binding.nibeuplink.internal.connector;
+
+import java.io.UnsupportedEncodingException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpStatus.Code;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.nibeuplink.config.NibeUplinkConfiguration;
+import org.openhab.binding.nibeuplink.handler.NibeUplinkHandler;
+import org.openhab.binding.nibeuplink.internal.command.Login;
+import org.openhab.binding.nibeuplink.internal.command.NibeUplinkCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles requests to the NibeUplink web interface. It manages authentication and wraps commands.
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class UplinkWebInterface {
+
+    private static final long LOGIN_WAIT_TIME = 1000;
+
+    private final Logger logger = LoggerFactory.getLogger(UplinkWebInterface.class);
+
+    /**
+     * Configuration of the bridge from
+     * {@link org.openhab.BoxHandler.fritzaha.handler.FritzAhaBridgeHandler}
+     */
+    private final NibeUplinkConfiguration config;
+
+    /**
+     * Bridge thing handler for updating thing status
+     */
+    private final NibeUplinkHandler uplinkHandler;
+
+    /**
+     * holds authentication status
+     */
+    private boolean authenticated = false;
+
+    /**
+     * HTTP client for asynchronous calls
+     */
+    private HttpClient asyncclient;
+    /**
+     * Maximum number of simultaneous asynchronous connections
+     */
+    private int asyncmaxconns = 20;
+
+    /**
+     * Constructor to set up interface
+     *
+     * @param config Bridge configuration
+     */
+    public UplinkWebInterface(NibeUplinkConfiguration config, NibeUplinkHandler handler) {
+        this.config = config;
+        this.uplinkHandler = handler;
+        asyncclient = new HttpClient(new SslContextFactory(true));
+        asyncclient.setMaxConnectionsPerDestination(asyncmaxconns);
+        try {
+            asyncclient.start();
+        } catch (Exception e) {
+            logger.error("Could not start HTTP Client");
+        }
+        authenticate();
+    }
+
+    /**
+     * executes any command provided by parameter
+     *
+     * @param command
+     */
+    public void executeCommand(NibeUplinkCommand command) {
+        if (!isAuthenticated()) {
+            authenticate();
+            try {
+                Thread.sleep(LOGIN_WAIT_TIME);
+            } catch (InterruptedException e) {
+                assert true;
+            }
+        }
+
+        if (isAuthenticated()) {
+
+            StatusUpdateListener statusUpdater = new StatusUpdateListener() {
+
+                @Override
+                public void update(CommunicationStatus status) {
+                    if (status.getHttpCode().equals(Code.SERVICE_UNAVAILABLE)) {
```

Comment:
In general equal methods should be used with the static value on the left side: `Code.SERVICE_UNAVAILABLE.equals(...)` This avoid possible null pointer exceptions if variable used on the left side would be null.

Reply:
fixed it. Although for readybility I prefer the variable to be checked to be on the left but you are absolutely right, it could cause NPEs